unix: switch default build to use mbedtls (not axtls)

dpgeorge opened the pull request:
This PR:
- adds custom mbedtls config to the unix port which matches closely the bare-metal configs (eg stm32, rp2)
- switches unix to build with mbedtls by default, instead of axtls

Switching from axtls to mbedtls increases the unix build by about 134k on x64 architectures.

The reasons to make this switch are:
1. axtls is not being maintained/updated, mbedtls is.
2. So CI and tests can run against mbedtls, which is now the main TLS library used by the ports (eg stm32, rp2, mimxrt, esp32).  Only esp8266 uses axtls.

dpgeorge committed:
unix/mpconfigport: Make MICROPY_EVENT_POLL_HOOK call usleep directly.

So that MICROPY_EVENT_POLL_HOOK can be used without including py/mphal.h.

Signed-off-by: Damien George <damien@micropython.org>

dpgeorge committed:
unix/mbedtls: Add custom config for mbedtls.

Based on existing bare-metal config from stm32.  Also uses shorter error
messages from lib/mbedtls_errors.

Signed-off-by: Damien George <damien@micropython.org>

dpgeorge committed:
unix/mpconfigport: Switch ussl from axtls to mbedtls.

The reasons to make this switch are:
- The axtls library is not being maintained/updated, mbedtls is.
- So CI and tests can run against mbedtls, which is now the main TLS
  library used by the ports (eg stm32, rp2, mimxrt, esp32). Only esp8266
  uses axtls.

Increases unix standard build on x86-64 by about 89000 bytes.

Signed-off-by: Damien George <damien@micropython.org>

dpgeorge committed:
tests/extmod/ussl_basic: Make test run on axtls and mbedtls.

Fixes issue #4364.

Signed-off-by: Damien George <damien@micropython.org>

Carglglz commented:
I've tested these changes locally and added #8252 on top and now all `test/multi_net/ssl*.py` are properly passing, although I had to change `key`, `cert` in `ssl_data.py` to the ones I generated in #8252. (I enabled PEM parse)

Here is the full config diff:
```diff
diff --git a/ports/unix/mbedtls/mbedtls_config.h b/ports/unix/mbedtls/mbedtls_config.h
index 1b6b4d14c..1cdf2770c 100644
--- a/ports/unix/mbedtls/mbedtls_config.h
+++ b/ports/unix/mbedtls/mbedtls_config.h
@@ -49,6 +49,7 @@
 #define MBEDTLS_ECP_NIST_OPTIM
 #define MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
 #define MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
+#define MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 #define MBEDTLS_NO_PLATFORM_ENTROPY
 #define MBEDTLS_PKCS1_V15
 #define MBEDTLS_SHA256_SMALLER
@@ -65,10 +66,12 @@
 // Enable mbedtls modules
 #define MBEDTLS_AES_C
 #define MBEDTLS_ASN1_PARSE_C
+#define MBEDTLS_ASN1_WRITE_C
 #define MBEDTLS_BIGNUM_C
 #define MBEDTLS_CIPHER_C
 #define MBEDTLS_CTR_DRBG_C
 #define MBEDTLS_ECDH_C
+#define MBEDTLS_ECDSA_C
 #define MBEDTLS_ECP_C
 #define MBEDTLS_ENTROPY_C
 #define MBEDTLS_ERROR_C
@@ -91,7 +94,12 @@
 #define MBEDTLS_TIMING_C
 #define MBEDTLS_TLS_DEFAULT_ALLOW_SHA1_IN_KEY_EXCHANGE
 #define MBEDTLS_X509_CRT_PARSE_C
+#define MBEDTLS_X509_CRL_PARSE_C
+#define MBEDTLS_X509_CSR_PARSE_C
 #define MBEDTLS_X509_USE_C
+#define MBEDTLS_BASE64_C
+#define MBEDTLS_PEM_PARSE_C
+
```
(I enabled `MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` so I could test unix->esp32 and works both ways too👍🏼 .)

Also not sure if there is a bug in `tests/run-multitest.py` 
```diff
diff --git a/tests/run-multitests.py b/tests/run-multitests.py
index d8a4a48fa..6dad04c74 100755
--- a/tests/run-multitests.py
+++ b/tests/run-multitests.py
@@ -36,6 +36,8 @@ class multitest:
             sys.stdout.flush()
         except AttributeError:
             pass
+        except OSError:
+            pass
     @staticmethod
     def skip():
         print("SKIP")
```
Otherwise `OSError: 45`  in flush makes the tests fail.
```
+Traceback (most recent call last):
+  File "<stdin>", line 117, in <module>
+  File "<stdin>", line 42, in instance0
+  File "<stdin>", line 98, in globals
+  File "<stdin>", line 72, in flush
+OSError: 45
```
🤷🏼

dpgeorge commented:
I trimmed down the mbedtls options a bit to match stm32.  The change in code size is now about +94000 bytes.  And the tests now pass.

dpgeorge committed:
unix/Makefile: Only include axtls in GIT_SUBMODULES list when enabled.

Signed-off-by: Damien George <damien@micropython.org>

dpgeorge committed:
extmod: Always use custom mbedtls error message code.

All ports that use mbedtls use the custom error messages in
mp_mbedtls_errors.c.  This commit simplifies the build so that ports don't
need to explicitly add this file, it's now used by default when mbedtls is
enabled.

Signed-off-by: Damien George <damien@micropython.org>

dpgeorge committed:
examples/embedding: Remove obsolete axtls build target.

axtls is now built as part of the standard make process, using rules in
extmod/extmod.mk.

Signed-off-by: Damien George <damien@micropython.org>

dpgeorge commented:
I made unix use the more compact mbedtls error message strings, and it now adds +89000 bytes to the executable, compared to axtls.
